uniffi_bindgen/src/bindings/gecko_js/templates/SharedHeaderTemplate.h

```diff
@@ -29,13 +29,17 @@ namespace dom {
 namespace {{ context.detail_name() }} {
 
 {% for e in ci.iter_enum_definitions() %}
+{% if e.has_associated_data() %}
+MOZ_STATIC_ASSERT(false, "Sorry the gecko-js backend does not yet support enums with associated data");
+{% else %}
 template <>
-struct ViaFfi<{{ e.name()|class_name_cpp(context) }}, uint32_t> {
-  [[nodiscard]] static bool Lift(const uint32_t& aLowered, {{ e.name()|class_name_cpp(context) }}& aLifted) {
-    switch (aLowered) {
+struct Serializable<{{ e.name()|class_name_cpp(context) }}> {
```

Comment:
This changes the way enums are passed over the FFI, even enums without any associated data. I've done my best to update this and will see if I can try it out in mozilla-central following the instructions in [this doc](https://docs.google.com/document/d/1JnSruKgfKDcfIuiMirEIhum4NOWYlo58sCXPoxtC21g/) but I don't have high confidence in my skills in this area.

Reply:
I was able to integrate this into m-c and have it compile, then fail on a linker error. So I guess that's a good sign.